lib/log.js

```diff
@@ -12,6 +13,53 @@ const logConfig = config.get('log')
 const StatsDCollector = require('./metrics/statsd')
 const unbuffer = require('./crypto/butil').unbuffer
 
+const userAgent = require('./userAgent')
+const crypto = require('crypto')
+const Amplitude = require('amplitude')
+var amplitude = new Amplitude(config.get('amplitude.apiKey'))
+
+var amplitudeLastSeenTimestamp = 0
+var amplitudeEventCounter = 0
+var amplitudeHmacKey = config.get('amplitude.hmacKey')
+
+function nextAmplitudeEventId(data) {
+  if (data.time > amplitudeLastSeenTimestamp) {
+    amplitudeLastSeenTimestamp = data.time
+    amplitudeEventCounter = 0
+  }
+  return ++amplitudeEventCounter
+}
+
+function anonymizeForAmplitude(id) {
+  return crypto.createHmac('sha256', amplitudeHmacKey)
+               .update(id)
+               .digest('hex')
+}
+
+function sendToAmplitude(data) {
+  if (! data.time) {
+    data.time = Date.now()
+  }
+  userAgent.call(data, data.userAgent, this)
```

Comment:
With apologies for poking my nose in at a WIP PR, but if we pulled this up a level to `lib/metrics/events.js` we might not need to call `userAgent` in most cases, because the parsed result already exists on `request.auth.credentials` for routes that take a session token. And the same change would also help out below...